clang_bind/cmake_frontend.py

```diff
@@ -0,0 +1,162 @@
+import os
+import json
+
+
+class Codemodel:
+    def __init__(self, reply_dir):
+        self.reply_dir = reply_dir
+        self.codemodel = {}
+        self.codemodel_targets = {}
+
+    def set_codemodel(self, codemodel_file=None):
+        if not codemodel_file:
+            for file in os.listdir(self.reply_dir):
+                if file.startswith("codemodel"):
+                    codemodel_file = os.path.join(self.reply_dir, file)
+        with open(codemodel_file) as f:
+            self.codemodel = json.load(f)
+
+    def get_codemodel(self):
+        if not self.codemodel:
+            self.set_codemodel()
+        return self.codemodel
+
+    def set_codemodel_targets(self, codemodel=None):
```

Comment:
Which function is the user facing API? Let's prepend the others with _

Having optional input in several places makes me try to reduce the places where the user can change it

Reply:
Removed the class taking into account the suggestion below.